### DISCUSSION TITLE
Confusing problem with links to files from README.md

### DISCUSSION THREAD

**Discussion by ekera** [community/community#69959] [Repositories]
### Select Topic Area

Bug

### Body

In my repository [https://github.com/ekera/regevnum](https://github.com/ekera/regevnum), in the <code>README.md</code> file that is automatically displayed below the listing of the root directory, some of the links are broken, and I have a hard time understanding why.

- For example, some of the four links to <code>simulator.sage</code> direct to "https://github.com/ekera/simulator.sage", whereas others direct to "https://github.com/ekera/regevnum/blob/main/simulator.sage", where the latter is correct.

   When I view the source for the <code>README.md</code> file, all of these links are identical, yet they seemingly direct to different URLs when <code>README.md</code> is rendered. I do not understand why. Am I missing something trivial, or is this actually a bug?

- If I click <code>README.md</code>, i.e. if I go to <code>https://github.com/ekera/regevnum/blob/main/README.md</code>, then all four links direct to "https://github.com/ekera/regevnum/blob/main/simulator.sage" which is correct.

- Also, when I view my repository in the GitHub app on my iPhone, all four links direct to the correct path.

**Comment by mohit-nagaraj** [comment]
Maybe what's likely happening is you're experiencing a discrepancy in link behavior when viewing the README.md file in your GitHub repository. This might be due to how relative and absolute paths are interpreted in different contexts. 

There's a key difference when viewing the README.md file directly in the repository (https://github.com/ekera/regevnum/blob/main/README.md) as opposed to viewing it while exploring the repository (https://github.com/ekera/regevnum). In the former case, the links are resolved as relative to the README.md file itself.

Try once using absolute URLs for links. For example:
![image](https://github.com/community/community/assets/105775899/3b406179-ad43-4575-83ec-29e6532e416f)

- Instead of `[simulator.sage](simulator.sage)`, use `[simulator.sage](https://github.com/ekera/regevnum/blob/main/simulator.sage)`.

By using absolute URLs, you ensure that the links work correctly in all contexts and eliminate any ambiguity related to relative paths.

I hope this helps clarify the issue.  If you found this information helpful, feel free to accept the answer. 😊

**Reply by ekera** [reply]
> Maybe what's likely happening is you're experiencing a discrepancy in link behavior when viewing the README.md file in your GitHub repository. This might be due to how relative and absolute paths are interpreted in different contexts. 
> 
> There's a key difference when viewing the README.md file directly in the repository (https://github.com/ekera/regevnum/blob/main/README.md) as opposed to viewing it while exploring the repository (https://github.com/ekera/regevnum). In the former case, the links are resolved as relative to the README.md file itself.

Yes, I am aware of this. But as I wrote, the issue is that multiple occurrences of the same link in <code>README.md</code> are seemingly rendered differently, resulting in URLs that are both correct and broken. This when viewing the root listing of the repository, and the <code>README.md</code> file as rendered below the listing. I suspect that this is a bug.

I do not want to use absolute links, since this would not work if I move or rename the repository, or add more branches, etc.
